src/components/ValueSelectorCard/ValueSelectorCard.js

```diff
@@ -0,0 +1,338 @@
+// @flow
+/*
+    Pillar Wallet: the personal data locker
+    Copyright (C) 2019 Stiftung Pillar Project
+
+    This program is free software; you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation; either version 2 of the License, or
+    (at your option) any later version.
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License along
+    with this program; if not, write to the Free Software Foundation, Inc.,
+    51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+*/
+
+import * as React from 'react';
+import styled from 'styled-components/native';
+import t from 'tcomb-form-native';
+import isEmpty from 'lodash.isempty';
+import { connect } from 'react-redux';
+import { createStructuredSelector } from 'reselect';
+import get from 'lodash.get';
+
+import ShadowedCard from 'components/ShadowedCard';
+import { BaseText } from 'components/Typography';
+
+import type { Assets, Balances, Rates } from 'models/Asset';
+import type { RootReducerState } from 'reducers/rootReducer';
+import type { TransactionFeeInfo } from 'models/Transaction';
+import type { FormSelector } from 'models/TextInput';
+
+import { formatAmount, formatFiat } from 'utils/common';
+import { spacing } from 'utils/variables';
+import { getBalance, getRate, sortAssets, calculateMaxAmount } from 'utils/assets';
+import { themedColors } from 'utils/themes';
+import { SelectorInputTemplate, selectorStructure, inputFormatter, inputParser } from 'utils/formHelpers';
+import { getFormattedBalanceInFiat } from 'screens/Exchange/utils';
+
+import { accountBalancesSelector } from 'selectors/balances';
+import { accountAssetsSelector } from 'selectors/assets';
+import { defaultFiatCurrency } from 'constants/assetsConstants';
+
+
+type Props = {
```

Comment:
Can we please have a preselectedValue option, as well as a prop flag to make assets selectable or fixed for the preselectedAsset only

Reply:
Also we would probably need a hasError callback from the Parent component to disable/enable buttons accordingly

Reply:
And the last needed functionality would be a custom balance label and value to be set as a prop
https://app.zeplin.io/project/5db9b5d14c30ae56faaa3030/screen/5ee37a8ce393b3999399ac8a